src/components/nano/ModalChangeAddress.js

```diff
@@ -0,0 +1,171 @@
+  const oldAddress = nanoContracts[props.nanoContractID].address;
+
+  // {number} step The modal has 3 steps: 0 is the warning screen, 1 is the address list and 2 is the confirmation
+  const [step, setStep] = useState(0);
+  // {Object} newAddress New nano contract selected address data {'address': string, 'index': number}
+  const [newAddress, setNewAddress] = useState(null);
+  // {number} oldAddressIndex the index of the old selected address. Need to be as a state because of the async/await call to the wallet method
+  const [oldAddressIndex, setOldAddressIndex] = useState(null);
+
+  useEffect(() => {
+    $('#changeAddressModal').modal('show');
+    $('#changeAddressModal').on('hidden.bs.modal', (e) => {
+      setStep(0);
+      // We always need to call on close when using global context modal
+      props.onClose();
+    })
+
+    async function getAndSetOldAddressIndex() {
+      const oldAddressIndex = await wallet.getAddressIndex(oldAddress);
+      setOldAddressIndex(oldAddressIndex);
+    }
+
+    getAndSetOldAddressIndex();
+
+    return () => {
+      $('#changeAddressModal').modal('hide');
+      // Removing all event listeners
+      $('#changeAddressModal').off();
+    };
+
+  }, []);
+
+  /**
+   * Called when user clicks to continue after reading the warning message in step 0
+   *
+   * @param {Object} e Event emitted on click
+   */
+  const goToStep1 = (e) => {
+    e.preventDefault();
+    setStep(1);
+  }
+
+  /**
+   * Called when user clicks in one of the addresses in step 1
+   *
+   * @param {string} address Address clicked
+   */
+  const onAddressClick = async (address) => {
+    const newAddressIndex = await wallet.getAddressIndex(address);
+    setNewAddress({ address, index: newAddressIndex});
+    setStep(2);
+  }
+
+  /**
+   * Called when user clicks to change the address in step 2
+   *
+   * @param {Object} e Event emitted on click
+   */
+  const executeChange = async (e) => {
+    e.preventDefault();
+    dispatch(editAddressNC(props.nanoContractID, newAddress.address));
+    await wallet.storage.updateNanoContractRegisteredAddress(props.nanoContractID, newAddress.address);
+    props.onClose();
+  }
+
+  const renderSteps = () => {
+    switch (step) {
+      case 0:
+        return renderWarning();
+      case 1:
+        return renderAddressList();
+      case 2:
+        return renderConfirmation();
+      default:
+        return null;
+    }
+  }
+
+  const renderWarning = () => {
+    return (
+      <div>
+        <p>{t`You can have only one address associated with a nano contract at a time. If you decide to change your address, any action that you execute in this nano contract will be associated with this new address.`}</p>
+        <p>{t`It's important that you remember which address you are using to execute the nano contract methods.`}</p>
+        <p><strong>{t`Please continue only if you understand your risks and know what you are doing.`}</strong></p>
```

Comment:
question(non-blocking): Should we consider adding something like "This could result in the loss of funds" warning, for legal reasons?

Reply:
But this couldn't result in loss of funds, it's just a matter of seeing the transactions correctly in the expected address.